src/na/zoned/decomposer/NativeGateDecomposer.cpp

```
@@ -0,0 +1,726 @@
+#include <variant>
+#include <vector>
+
+namespace na::zoned {
+namespace {
+template <class... Ts> struct overloads : Ts... {
+  using Ts::operator()...;
+};
+} // namespace
+NativeGateDecomposer::NativeGateDecomposer(const Architecture&,
+                                           const Config& config) {
+  config_ = config;
```

Cppcheck (2.21.0)

[performance] Variable 'config_' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)